### PR TITLE
[helm] added support of annotations for services

### PR DESCRIPTION
**Pull request by wulfthimm**
Annotations to services can be made using the variable `serviceAnnotations` in components.

```
components:
    serviceAnnotations: {}
```

### CODE REVIEW

**Comment by csweichel** [comment]
Could you elaborate a bit on the use-case? Why do we need this ability?

Irregardless, I think we should not add the empty map to all components unless a component actually needs an annotation. Instead we should check in the template if the map is set.

File: chart/values.yaml [hunk 1]
```diff
@@ -308,11 +318,11 @@ components:
       http:
         expose: true
         containerPort: 80
-        nodePort: 31080
+        # nodePort: 31080
```
Comment (csweichel) [inline chart/values.yaml:321]: Why disable the `nodePort`? This is what makes (or at least used to) the load balancer work on gitpod.io.

File: chart/values.yaml [hunk 2]
```diff
@@ -308,11 +318,11 @@ components:
...
       https:
         expose: true
         containerPort: 443
-        nodePort: 31443
+        # nodePort: 31443
```
Comment (csweichel) [inline chart/values.yaml:325]: Why disable the `nodePort`? This is what makes (or at least used to) the load balancer work on gitpod.io.

File: chart/values.yaml (hunk 1 above)
Comment (wulfthimm) [reply chart/values.yaml]: There is a conflict when the service type "ClusterIP" is used. Therefore `nodePort` has to be excluded from the default `values.yaml`. But this is not directly related to my commit. I have removed it.

File: chart/values.yaml (hunk 2 above)
Comment (wulfthimm) [reply chart/values.yaml]: There is a conflict when the service type "ClusterIP" is used. Therefore `nodePort` has to be excluded from the default `values.yaml`. But this is not directly related to my commit. I have removed it.

**Comment by csweichel** [comment]
> I prefer having every possible configuration value in the `values.yaml`. But I have changed it nonetheless.

We already don't have all options everywhere, but rely on defaults instead. E.g. the imageName defaults to components name and isn't explicitly set again. Same goes for the config-map name.

I agree that it's good to document the possibilities, but we need sensible defaults to not add too much noise.

File: chart/templates/_default.yaml [hunk 3]
```diff
@@ -18,6 +18,12 @@ metadata:
     {{- range $k, $v := $comp.svcLabels }}
     {{ $k }}: {{ $v }}
     {{- end }}
+{{- if $comp.serviceAnnotations }}
+{{- with $comp.serviceAnnotations }}
+  annotations:
+{{ toYaml . | indent 4 }}
+{{- end }}
+{{- end }}
```
Comment (csweichel) [inline chart/templates/_default.yaml:26]:
```suggestion
{{- if $comp.serviceAnnotations }}
  annotations:
{{ toYaml $comp.serviceAnnotations | indent 4 }}
{{- end }}
```

**Comment by ncalandra** [comment]
>
> Could you elaborate a bit on the use-case? Why do we need this ability?
> 

Service annotations allow for vendor specific configuration settings such as internal load balancers: https://kubernetes.io/docs/concepts/services-networking/service/#internal-load-balancer.

**Review by csweichel — APPROVED** [review]
LGTM

**Pull request merged**